Mergin/plugin.py

```diff
@@ -316,8 +322,11 @@ def sync_project(self):
             elif isinstance(dlg.exception, ClientError):
                 QMessageBox.critical(None, "Project sync", "Client error: " + str(dlg.exception))
             else:
-                # TODO: this should not happen normally - "known" exceptions should be recognized
-                msg = "Failed to upload changes!\n\n{}".format(str(dlg.exception))
+                msg = "Failed to sync project {} due to an unhandled exception.\n\n" \
+                      "This should not happen, please report the problem here:\n" \
+                      "https://github.com/lutraconsulting/qgis-mergin-plugin/issues/new\n\n" \
```

Comment:
I do not like it is 3 times the same text and link ,... maybe some helper function, so you do not need to maintain it on various places around?

Reply:
fixed in 58bf651